src/barfturn.ts

```diff
@@ -0,0 +1,270 @@
+import {
+  availableAmount,
+  canAdventure,
+  currentRound,
+  inebrietyLimit,
+  myAdventures,
+  myInebriety,
+  myLevel,
+  runChoice,
+  runCombat,
+  totalTurnsPlayed,
+  toUrl,
+  use,
+  useFamiliar,
+  visitUrl,
+} from "kolmafia";
+import {
+  $effect,
+  $item,
+  $items,
+  $location,
+  $monster,
+  $skill,
+  adventureMacro,
+  adventureMacroAuto,
+  clamp,
+  Counter,
+  ensureEffect,
+  get,
+  getModifier,
+  have,
+  Requirement,
+  SourceTerminal,
+} from "libram";
+import { Macro, withMacro } from "./combat";
+import { completeBarfQuest } from "./dailies";
+import { estimatedTurns } from "./embezzler";
+import { barfFamiliar, freeFightFamiliar, meatFamiliar } from "./familiar";
+import {
+  embezzlerLog,
+  globalOptions,
+  kramcoGuaranteed,
+  safeRestore,
+  setChoice,
+  steveAdventures,
+} from "./lib";
+import { meatMood } from "./mood";
+import {
+  familiarWaterBreathingEquipment,
+  freeFightOutfit,
+  meatOutfit,
+  tryFillLatte,
+  waterBreathingEquipment,
+} from "./outfit";
+import { determineDraggableZoneAndEnsureAccess, digitizedMonstersRemaining } from "./wanderer";
+const embezzler = $monster`Knob Goblin Embezzler`;
+
+function embezzlerPrep(requirements?: Requirement) {
+  useFamiliar(meatFamiliar());
+  meatOutfit(true, requirements);
+}
+
+function freeFightPrep(requirements?: Requirement) {
+  useFamiliar(freeFightFamiliar(true));
+  freeFightOutfit(requirements);
+}
+
+const sober = () => myInebriety() <= inebrietyLimit();
+
+type Turn = {
+  name: string;
+  available: () => boolean;
+  perform: () => boolean;
+};
+
+const turns: Turn[] = [
```

Comment:
okay so this is just like, a big ol list of turns where you want to do them, and you want to check availability? 

while i think it makes some sense i have a few questions. okay, more than a few questions.

- does order matter here? was this ordered specifically or just randomly?
- "perform" feels like a weird way to name that part of it; runTurn or runAdventure feels better, "perform" just seems like a technically accurate but not fully definied thing that adds a level of obfuscation
- i would probably add a docstring explaining the order piece and how to add new things to this in the future, as you will do this pretty often i expect

Reply:
I will probably attach most of the documentation to the `Turn` type, but that tracks. I'll run some language by you.